### PR TITLE
add context to Parse.Object.save

### PR DESCRIPTION
**Pull request by mtrezza**
Exposes the `context` object in `Parse.Object.save()` so that parameters can be made accessible in Cloud Code triggers `beforeSave`, `afterSave`.

Example:

```
const obj = new TestObject();
await obj.save(null, {context: {a: "a"}});

Parse.Cloud.beforeSave("TestObject", req => {
    console.log(req.context.a);
});

Parse.Cloud.afterSave("TestObject", req => {
    console.log(req.context.a);
});  
```
---
Don't merge until

- [ ] Concept agreed on in https://github.com/parse-community/parse-server/issues/6459
- [x] JS doc added in code for ‘context’

### CODE REVIEW

**Comment by mtrezza** [comment]
The context is transmitted as `_context` property in REST request.

- I did not find an existing similar use of internal properties in the REST request, but according to the [docs](https://docs.parseplatform.org/js/guide/#data-storage) a field name must start with a letter, so there cannot be a conflict with an object property:

> Key names must contain only numbers, letters, and underscore, and must start with a letter. 

So using `_context` as internal property in a REST request seems OK, it is also used for internal field names such as `_created_at` in the DB.

**Comment by mtrezza** [comment]
This unidirectionally exposes the internal trigger `context` to the client.
Should this be restricted to masterKey or to within Cloud Code?

@dplewis @acinader any thoughts?

File: src/__tests__/ParseObject-test.js [hunk 1]
```diff
@@ -1219,6 +1219,26 @@ describe('ParseObject', () => {
     });
   });
 
+  it('accepts context on save', async () => {
```
Comment (dplewis) [inline src/__tests__/ParseObject-test.js:1222]: Looks good, can you add an integration test?

Comment (mtrezza) [reply src/__tests__/ParseObject-test.js:1222]: I thought about this, but this PR requires a PR in parse server and vice versa. How would I set up an integration test?

Comment (mtrezza) [reply src/__tests__/ParseObject-test.js:1222]: Btw, the PR in parse server has a test case which tests the integration.

Comment (dplewis) [reply src/__tests__/ParseObject-test.js:1222]: Oh right I didn't see that.

Comment (mtrezza) [reply src/__tests__/ParseObject-test.js:1222]: No worries, let me know if anything else to take a look at...

File: src/ParseObject.js [hunk 2]
```diff
@@ -1251,6 +1252,9 @@ class ParseObject {
     if (options.hasOwnProperty('installationId') && typeof options.installationId === 'string') {
       saveOptions.installationId = options.installationId;
     }
+    if (options.hasOwnProperty('context') && typeof options.context === 'object') {
+      saveOptions.context = options.context;
```
Comment (dplewis) [inline src/ParseObject.js:1256]: Can you add documentation above for options?

Comment (mtrezza) [reply src/ParseObject.js:1256]: Ah right, got that as TODO above.

Comment (mtrezza) [reply src/ParseObject.js:1258]: done

**Comment by mtrezza** [comment]
Thanks! Would be great if you could make a release with parse server release that will merge https://github.com/parse-community/parse-server/pull/6626.

**Pull request merged**